src/NServiceBus.Core/Transports/Msmq/Config/MsmqTransportConfigurator.cs

```diff
@@ -18,6 +22,54 @@ public class MsmqTransportConfigurator : ConfigureTransport
         internal MsmqTransportConfigurator()
         {
             DependsOn<UnicastBus>();
+            RegisterStartupTask<CheckQueuePermissions>();
+        }
+
+        class CheckQueuePermissions : FeatureStartupTask
+        {
+            readonly IBuilder builder;
+
+            public CheckQueuePermissions(IBuilder builder)
+            {
+                this.builder = builder;
+            }
+
+            protected override void OnStart()
+            {
+                var queueCreators = builder.BuildAll<IWantQueueCreated>();
+
+                foreach (IWantQueueCreated creator in queueCreators)
+                {
+                    var msmqAddress = MsmqAddress.Parse(creator.Address);
+                    var queuePath = MsmqQueueCreator.GetFullPathWithoutPrefix(msmqAddress);
+
+                    if (MessageQueue.Exists(queuePath))
+                    {
+                        using (var messageQueue = new MessageQueue(queuePath))
+                        {
+                            WarnIfPublicAccess(messageQueue);
+                        }
+                    }
+                }
+            }
+
+            void WarnIfPublicAccess(MessageQueue queue)
+            {
+                MessageQueueAccessRights? everyoneRights, anonymousRights;
+
+                queue.TryGetPermissions(MsmqConstants.LocalAnonymousLogonName, out anonymousRights);
+                queue.TryGetPermissions(MsmqConstants.LocalEveryoneGroupName, out everyoneRights);
+
+                if (anonymousRights.HasValue && everyoneRights.HasValue)
+                {
+                    Logger.WarnFormat("Queue [{0}] is running with [{1}] and [{2}] permissions. Consider changing those, as required", 
```

Comment:
@johnsimons good idea, one of the tasks here is fix the docs, but I think I read somewhere we don't like to use links in messages, because they might become 404?